crates/core/component/dex/src/lib.rs

```diff
@@ -1,11 +1,15 @@
 #![deny(clippy::unwrap_used)]
 // Requires nightly.
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
-#[cfg(feature = "component")]
-pub mod component;
-pub mod event;
+
+cfg_if::cfg_if! {
+    if #[cfg(feature="component")] {
+        pub mod component;
+        pub mod event;
```

Comment:
Fair point, we should align imports and not the exports. But I'm confused about why the proposed breakage isn't detected at build time: I tried cherry-picking this commit onto the dex PRs you reference (#4894) and I was able to build `pindexer` just fine. That's not the same as pindexer actually working, of course, but I'd expect mucking around with the event interface would break things early.